Analysis/ingrid/private/likelihood_utils.nim

```diff
@@ -72,6 +72,61 @@ proc getInterpolatedWideDf*(df: DataFrame, num = 1000): DataFrame =
     dfLoc["Variable"] = constantColumn(tup[0][1].toStr, dfLoc.len)
     result.add dfLoc
 
+proc morphToA*(df: DataFrame, lineEnergies: seq, energy: float, offset = 1): (Tensor[int], Tensor[float]) =
+  ## generates a distribution for the appropriate energy `energy` between the
+  ## distribution below and above `energy` using linear interpolation
+  ## DF needs to have columns:
+  ## - "Hist": counts of distributions
+  ## - "Bins": bin edges of distributions
+  ## - "Dset": name of the target filter combination
+
+  let idx = max(lineEnergies.lowerBound(energy) - 1, 0)
+
+  # need idx and idx+offset
+
+  let refLow = int(lineEnergies[idx])
+  let refHigh = int(lineEnergies[idx+offset])
+ 
+
+  let refLowT = df[$refLow, float]
+  let refHighT = df[$refHigh, float]
+
+  let bins = df["bin", int]
+
+
+  var res = zeros[float](refLowT.size.int)
+  # walk over each bin and compute linear interpolation between
+  let Ediff = abs(lineEnergies[idx] - lineEnergies[idx+offset])
+  for i in 0 ..< bins.size:
+    res[i] = refLowT[i] * (1 - (abs(lineEnergies[idx] - energy)) / Ediff) +
+      refHighT[i] * (1 - (abs(lineEnergies[idx+offset] - energy)) / Ediff)
+  result = (bins, res)
+
+proc getInterpolatedDfToA*(df: DataFrame, lineEnergies: seq, num = 100): DataFrame =
+  ## returns a DF with `num` interpolated distributions using next neighbors
+  ## for linear interpolation
+  echo "enter interpolation"
+  let energies = linspace(lineEnergies[0], lineEnergies[^1], num)
+  var test: int
+  var dfLoc = newDataFrame()
+  var lastBins = zeros[int](0)
+  result = newDataFrame()
+  for idx, E in energies:
+    let (bins, res) = morphToA(df, lineEnergies, E, offset = 1)
+
+    block Sanity:
+      # really the same bins in all Target/Filter combinations? Should be, but check!
+      if lastBins.size > 0:
+        doAssert bins == lastBins
+      lastBins = bins
+    let suffix = "_" & $idx
+    dfLoc["Hist" & $suffix] = res
+
+  dfLoc["Bins"] = lastBins
+  #echo dfLoc  
+  result.add dfLoc
```

Comment:
Similarly to the above procedure, the actual logic can also be refactored out into a common procedure for regular / ToA cases. Just needs an argument that hands the internal morphing procedure to call. Again, I can do that later.